carbon-design-system/carbon
# fix(react): update radio button group state initialization

Closes #9041 

This updates how we initialize our `selected` state value from a `||` expression to a ternary that checks if the value is undefined. When using `||`, the state would be incorrectly initialized if the value was false-y but not undefined (like the value `0`).

I also took this small change to convert some of the tests for `RadioButtonGroup` to RTL. It also updates the `jest-dom` matchers to be brought in automatically.

#### Changelog

**New**

**Changed**

- Update `RadioButtonGroup.js` to correctly initialize `selected` state
- Update `RadioButtonGroup-test.js` file to start using RTL
- Update `jest-config-carbon` to automatically bring in `jest-dom` matchers

**Removed**

#### Testing / Reviewing

- Review the test case and make sure it mirrors the one brought up in #9041 
- Verify test passes
- Verify component continues to work in Storybook
